host-src/tool/syscalls.c

```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
     out_size = recv_uint();
 
     if (in_size)
-	recv_data(gdb_buf, in_size > GDBBUFSIZE ? GDBBUFSIZE : in_size, 0);
+        recv_data(gdb_buf, in_size > GDBBUFSIZE ? GDBBUFSIZE : in_size, 0);
+
+#ifdef __MINGW32__
+    /* Winsock SOCKET is defined as an unsigned int, so -1 won't work here */
+    static SOCKET socket_fd = 0;
+
+    if (gdb_server_socket == INVALID_SOCKET) {
+#else
+    static int socket_fd = 0;
 
     if (gdb_server_socket < 0) {
-	send_uint(-1);
-	return;
+#endif
+        send_uint(-1);
+        return;
     }
 
-    if (socket_fd < 0) {
-	printf( "waiting for gdb client connection...\n" );
-	if ( (socket_fd = accept( gdb_server_socket, NULL, NULL )) < 0) {
-	    perror("error accepting gdb server connection");
-	    send_uint(-1);
-	    return;
-	}
+    if (socket_fd == 0) {
+        printf( "waiting for gdb client connection...\n" );
+        socket_fd = accept( gdb_server_socket, NULL, NULL );
+#ifdef __MINGW32__
+        if (socket_fd != INVALID_SOCKET)
```

Comment:
Is this line correct? It just doesn't seem like it is...

Reply:
@sizious

Reply:
Fixed